test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go

```diff
@@ -1348,6 +1348,163 @@ var _ = Describe("[rfe_id:28761][performance] Updating parameters in performance
+				if onlineCPUInt <= 8 {
+					Skip(fmt.Sprintf("This test needs more than 8 CPUs online to work correctly, current online CPUs are %s", onlineCPUCount))
+				}
+			}
+			//numaNode[node][coreId][core-siblings]
+			var numaCoreSiblings map[int]map[int][]int
+			for _, node := range workerRTNodes {
+				numaCoreSiblings, err = nodes.GetCoreSiblings(&node)
+			}
+			//Lets take reserved cpus from the middle of the cpu list
+			for key := range numaCoreSiblings {
+				cores := make([]int, 0)
+				for k := range numaCoreSiblings[key] {
+					cores = append(cores, k)
+				}
+				sort.Ints(cores)
+				middleCoreIds := cores[len(cores)/2]
+				siblings := nodes.GetCpuSiblings(numaCoreSiblings, middleCoreIds)
+				reserved = append(reserved, siblings...)
+			}
+			reservedCpus := strings.Join(reserved, ",")
+			for key := range numaCoreSiblings {
+				for k := range numaCoreSiblings[key] {
+					siblings := nodes.GetCpuSiblings(numaCoreSiblings, k)
+					isolated = append(isolated, siblings...)
+				}
+			}
+			isolatedCpus := strings.Join(isolated, ",")
+			// Update performance profile
+			reservedSet := performancev2.CPUSet(reservedCpus)
+			isolatedSet := performancev2.CPUSet(isolatedCpus)
+
+			By("Update reserved, isolated parameters")
+			profile.Spec.CPU = &performancev2.CPU{
+				Reserved: &reservedSet,
+				Isolated: &isolatedSet,
+			}
+
+			By("Updating the performance profile")
+			profiles.UpdateWithRetry(profile)
+
+			By("Applying changes in performance profile and waiting until mcp will start updating")
+			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+
+			By("Waiting for MCP being updated")
+			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+
+			//Check RPS Mask after profile is updated with New reserved Cpus
+			expectedRPSCPUs, err := cpuset.Parse(string(*profile.Spec.CPU.Reserved))
+			Expect(err).ToNot(HaveOccurred())
+			for _, node := range workerRTNodes {
+				// Verify the systemd RPS service uses the correct RPS mask
+				var maskContent string
+				cmd := []string{"cat", "/rootfs/etc/systemd/system/update-rps@.service"}
+				unitFileContents, err := nodes.ExecCommandOnNode(cmd, &node)
+				Expect(err).ToNot(HaveOccurred())
+				for _, line := range strings.Split(unitFileContents, "\n") {
+					if strings.Contains(line, "ExecStart=/usr/local/bin/set-rps-mask.sh") {
+						maskContent = line
+					}
+				}
+				rpsMaskContent := strings.TrimSuffix(maskContent, "\r")
+				Expect(len(strings.Split(rpsMaskContent, " "))).To(Equal(3), "systemd unit file doesn't have proper rpsmask")
+				serviceRPSCPUs := strings.Split(rpsMaskContent, " ")[2]
+				rpsCPUs, err := components.CPUMaskToCPUSet(serviceRPSCPUs)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(rpsCPUs).To(Equal(expectedRPSCPUs), "the service rps mask is different from the reserved CPUs")
+
+				// Verify all host network devices have the correct RPS mask
```

Comment:
We already have a similar test for that right? is it different than the old one?

Reply:
Yes its kind of similar where we are testing rps mask. The difference here, we are testing rps mask where we are modifying the reserved cpus and we are using cpus from the middle of the avaiable cpu id's.  Once the reserved cpus are updated we are checking if the rps mask got applied as per our cpus that we picked up for reserved.